tests/system/test_root_metronome.py

```diff
@@ -239,6 +239,28 @@ def test_docker_job():
         assert len(client.get_runs(job_id)) == 1
 
 
+def test_metronome_shutdown_with_no_extra_tasks():
+    """ Test for METRONOME-100 regression
+        When Metronome is restarted it incorrectly started another task for already running job run task.
+    """
+    client = metronome.create_client()
+    job_id = "metronome-shutdown-{}".format(uuid.uuid4().hex)
+    with job(job_no_schedule(job_id)):
+        # run a job before we shutdown Metronome
+        client.run_job(job_id)
+        time.sleep(2)
+        common.assert_one_job_run_with_one_task(client, job_id)
+
+        # restart metronome process
+        metronome_leader = shakedown.marathon_leader_ip()
+        shakedown.run_command_on_agent(metronome_leader, 'sudo systemctl restart dcos-metronome')
```

Comment:
it is unclear if this will provide a good test under a multi-master situation... I think it will but I haven't tested it.   I'm not sure if this condition happens at the point of becoming leader or at the point of startup (as leader).   However... in a multi-master environment... the `time_wait` below will return almost immediately without a full restart taking place as the standby metronome takes over leadership.

Reply:
good point, I'll test it in multi-master setup as well

Reply:
I tested it and it works ok. I think when leader takes over the new leader have to load the state from the ZK as there is no in-memory state in the stand-by leader. But correct me if I am wrong...

Btw. it would be nice to add /leader endpoint to metronome

Reply:
oh sorry, on multimaster the test PASSES which is weird... will have to look into this once more tomorrow

Reply:
I added a comment - I think right now there is no easy way how to make this work in multimaster setup